### PR TITLE
[Redesign] Update DetectionDecayOperator

### PR DESCRIPTION
**Pull request by objorkman**
* Update and connect `DetectionDecayOperator` to `DetectionOperator`
* Modify `DetectionOperator` to send dictionary messages containing the runtime and obstacles, instead of `ObstaclesMessage`

### CODE REVIEW

**Review by pschafhalter — APPROVED** [review]
LGTM. Could you fix the merge conflicts?

@sukritkalra, are you in favor of having the detection operator send dicts of obstacles and runtimes?

File: pylot/perception/detection/detection_decay_operator.py [hunk 1]
```diff
@@ -26,29 +23,14 @@ def __init__(self, obstacles_stream, map_stream, flags):
...
-
-        Returns:
-            :py:class:`erdos.WriteStream`: Stream on which the operator
-            publishes mAP accuracy results.
-        """
-        map_stream = erdos.WriteStream()
-        return [map_stream]
-
-    def on_ground_obstacles(self, msg, map_stream):
+    def on_data(self, context: OneInOneOutContext, data: erdos.Message):
```
Comment (pschafhalter) [inline pylot/perception/detection/detection_decay_operator.py:26]:
```suggestion
    def on_data(self, context: OneInOneOutContext, data: dict):
```

**Comment by sukritkalra** [comment]
Is the concern that we don't want to maintain an `Obstacles` type? Generally, for use-cases like these, I prefer `namedtuple`s instead, but that does require us to define an `Obstacles` or `DetectorOutput` namedtuple somewhere and type everything accordingly. 

Dictionaries require developers to check for contents passed using upstream operators, and might also create a bad programming habit of mutating the dictionaries in place and passing them along downstream.

**Review by sukritkalra — APPROVED** [review]
LGTM!

**Pull request merged**